docs/relationship_synthesis.md

```diff
@@ -0,0 +1,266 @@
+specified otherwise with `caseSensitive: true`. 
+Multiple conditions must all match for the rule to apply; there is no support for `OR` conditions.
+
+## Relationship
+
+The `relationship` section defines aspects related to the created relationship.
+
+### Expires
+
+The `expires` field allows configuring the duration for which the relationship should exist if it is not reported within that timeframe. 
+By default, this field is set to ***75 minutes*** (the default for the relationship platform). 
+The value for expires follows the `ISO-8601` format and must be between ***10 minutes*** and ***75 minutes***, which is the time interval 
+currently supported by the platform.
+
+Explicitly providing the expires field is recommended for clarity, even though the default is 75 minutes.
+
+### RelationshipType
+
+The `relationshipType` field specifies the type of relationship to create. Only predefined values from the 
+supported relationship types list are accepted.
+
+#### Source & Target Resolvers
+
+Resolvers define how the source and target GUIDs are determined. There are three types of resolvers:
+
+1. `extractGuid`: Extracts the GUID from an existing attribute in the telemetry.
+2. `buildGuid`: Constructs the GUID using various pieces of information from the telemetry.
+3. `lookupGuid`: Performs a lookup using candidates to find the matching GUID based on telemetry attributes. 
+
+##### extractGuid
+
+The `extractGuid` resolver is the simplest and allows specifying an attribute to use as the GUID.
+
+Example:
+
+```yaml
+  source:
+    extractGuid:
+      attribute: host.guid
+      entityType:
+        value: HOST
+  target:
+    extractGuid:
+      attribute: entityGuid
+      entityType:
+        attribute: nr.entityType
+```
+
+If the resolved GUID corresponds to an `INFRA-NA` GUID, the relationship platform needs to know the actual type behind the GUID. 
+This can be achieved by providing a second field that specifies a hardcoded value for the entity type 
+or retrieves it from an existing attribute.
+
+##### buildGuid
+
+The `buildGuid` resolver allows constructing the GUID from different pieces of information.
+
+Example:
+
+```yaml
+source:
+  buildGuid:
+    account:
+      attribute: accountId
+      # or lookup: yes
+    domain:
+      value: EXT
+    type:
+      value: SERVICE
+      valueInGuid: NA
+    identifier:
+      fragments:
+      - value: "k8s:"
+      - attribute: "service.name"
+      - value: ":pod:"
+      hashAlgorithm: FARM_HASH
+```
+
+The `account` field can be defined using an attribute in the telemetry or by specifying a lookup key. 
+When a lookup key is defined, it will be searched within the accounts of the trusted account for the current data point (proposal system).
+
+The domain and type fields can be either hardcoded values or attributes. The type field also allows setting the given value as the type in the 
+GUID using `valueInGuid`, which is useful for building `INFRA-NA` GUIDs.
```

Comment:
We are leaking INFRA-NA here which is an internal concept, not sure what to do about it. cc @naxhh

Reply:
yes, we can't do much sadly, we tried to hide it in some way but is not really possible.
Notice this is already exposed on the entity synthesis definitions too.

I would probably aim for the documentation to be about the generic use case and
have a very brief line explaining NA can be assigned that way

Reply:
Extracted INFRA-NA to its own section: https://github.com/newrelic/entity-definitions/pull/1138/commits/cfd617ee612b3b7b28edb5afe16b39458cf3cf9a